Sources/FunctionalTableData/CellConfigType.swift

```diff
@@ -63,3 +63,80 @@ public extension CellConfigType {
 		return type(of: self) == type(of: other)
 	}
 }
+
+public struct AnyCellConfigType: CellConfigType, Hashable {
+	public func dequeueCell(from tableView: UITableView, at indexPath: IndexPath) -> UITableViewCell {
+		return base.dequeueCell(from: tableView, at: indexPath)
+	}
+	
+	public static func ==(lhs: AnyCellConfigType, rhs: AnyCellConfigType) -> Bool {
+		return lhs.sectionKey == rhs.sectionKey && lhs.key == rhs.key && lhs.base.isSameKind(as: rhs.base) && lhs.base.isEqual(rhs.base)
```

Comment:
Any reason we can't we defer to the `isEqual` method here? I'd expect both `==` and `isEqual` methods to be equivalent.

```suggestion
		return lhs.isEqual(rhs)
```

Reply:
In this case, no.

Equality and hash have to use the same fields, otherwise the diffable data source will sometimes fail to apply the snapshot (with an exception).

We include the sectionKey here to tie each cell with its section to account for a cell with the same key and state in multiple sections. If we don't, diffable data sources just uses the last one it saw with the same hash. I think it stores sections and rows together in the same hashtable or something.

`AnyCellConfigType` is solely around for being Hashable, to satisfy the compiler and the requirements of diffable data sources. It should probably be internal, but there's no harm in making it public. For its `CellConfigType` methods, it should defer to its `base` instance.